File: LeetCode/JavaScript/#409 Longest Palindrome.js
/**
 * @param {string} s
 * @return {number}
 */
var longestPalindrome = function(s) {
    var res = {}
    var len = s.length;
    var ans = 0;
    for(var i in s){
      if(res[s[i]]){
        res[s[i]] +=1;
      }else{
        res[s[i]] = 1;
      }
    }
    for(var key in res){
      if(res[key]%2===0){
        ans +=res[key];
      }else{
        ans +=res[key]-1;
      }
    }
    if(len!==ans){
      ans +=1;
    }
    return ans;

};

File: LeetCode/Python/#409 Longest Palindrome.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        flag = True
        res = 0
        for i in set(s):
            temp =s.count(i)
            if temp%2==0:
                res += temp
            else:
                res += temp-1
                flag = False
        if not flag:
            return res+1
        else:
            return res


class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        lens = 0
        for i in set(s):
            if s.count(i)%2==0:
                lens +=s.count(i)
            else:
                lens +=s.count(i)-1

        if len(s)!=lens:
            lens +=1
        return lens